cryptotik/kraken.py

```diff
@@ -0,0 +1,227 @@
+
+    @classmethod
+    def get_markets(cls, filter=None):
+        '''Find supported markets on this exchange,
+            use <filter> if needed'''
+
+        r = cls.api(cls.url + "public/AssetPairs")
+        return [i for i in r]
+
+    @classmethod
+    def get_market_ticker(cls, pair):
+        '''returns simple current market status report'''
+        p = cls.format_pair(pair)
+        return cls.api(cls.url + "public/Ticker", 
+                params={'pair': p})[p]
+
+    @classmethod
+    def get_market_volume(cls, pair):
+        ''' return volume of last 24h'''
+
+        return cls.get_market_ticker(cls.format_pair(pair))['v'][1]
+
+    @classmethod
+    def get_market_trade_history(cls, pair, limit=10):
+        '''get market trade history'''
+
+        p = cls.format_pair(pair)
+        return cls.api(cls.url + "public/Trades", 
+                        params={'pair': p})[p][:limit]
+
+    @classmethod
+    def get_market_orders(cls, pair, limit=100):
+        '''return order book for the market'''
+
+        p = cls.format_pair(pair)
+        return cls.api(cls.url + "public/Depth", 
+                        params={'pair': p, 'count': limit})[p]
+
+    @classmethod
+    def get_market_spread(cls, pair):
+        '''return first buy order and first sell order'''
+
+        order_book = cls.get_market_orders(cls.format_pair(pair))
+
+        ask = order_book["asks"][0][0]
+        bid = order_book["bids"][0][0]
+
+        return Decimal(ask) - Decimal(bid)
+
+    @classmethod
+    def get_market_depth(cls, pair):
+        '''return sum of all bids and asks'''
+
+        order_book = cls.get_market_orders(cls.format_pair(pair))
+        asks = sum([Decimal(i[1]) for i in order_book['asks']])
+        bid = sum([Decimal(i[1]) for i in order_book['bids']])
+
+        return {"bids": bid, "asks": asks}
+
+    def __init__(self, apikey=None, secret=None, timeout=None):
+        '''initialize class'''
+
+        if apikey and secret:
+            self.apikey = apikey.encode('utf-8')
+            self.secret = secret.encode('utf-8')
+
+    try:
+        assert timeout is not None
+    except:
+        timeout = (8, 15)
+    
+    def get_nonce(self):
+        '''return nonce integer'''
+
+        return int(1000*time.time())
+
+    def private_api(self, url, params={}):
+        '''handles private api methods'''
+        
+        if not self.apikey or not self.secret:
+            raise ValueError("A Key and Secret needed!")
+
+        urlpath=url[22:]
+        data=params
+        data['nonce'] = self.get_nonce()
+        postdata = requests.compat.urlencode(data)
+        encoded = (str(data['nonce']) + postdata).encode()
+        message = urlpath.encode() + hashlib.sha256(encoded).digest()
+        signature = hmac.new(base64.b64decode(self.secret),
+                    message, hashlib.sha512)
+        sigdigest = base64.b64encode(signature.digest())
+        
+        result = requests.post(url, data=data, headers={
```

Comment:
I'd rather use requests.session for the private API.
https://github.com/peerchemist/cryptotik/blob/master/cryptotik/bittrex.py#L20